add-chat-push-notifications-auto-renew/SwiftPushTest/AppDelegate.swift

```diff
@@ -0,0 +1,267 @@
+        // Setup user token client and chat client
+        setupUserTokenClient()
+        setupChatClient()
+        
+        // Register for services
+        registerForPushNotifications()
+        registerBackgroundTasks()
+        
+        return true
+    }
+    
+    // MARK: - Setup Methods
+    
+    /// Sets up the UserTokenClient with token refresh logic.
+    private func setupUserTokenClient() {
+        userTokenClient = UserTokenClient(tokenIssuerURL: appSettings.tokenIssuerURL!)
+
+        tokenRefresher = { tokenCompletionHandler in
+            self.userTokenClient.getNewUserContext { success, error in
+                if success, let token = self.userTokenClient.getUserToken {
+                    tokenCompletionHandler(token, nil)
+                } else {
+                    tokenCompletionHandler(nil, error)
+                }
+            }
+        }
+    }
+        
+    /// Initializes and configures the ChatClient with token credential.
+    private func setupChatClient() {
+        guard let tokenRefresher = self.tokenRefresher else {
+            //Token refresher is not set up
+            return
+        }
+
+        do {
+            let refreshOptions = CommunicationTokenRefreshOptions(
+                initialToken: appSettings.initialToken,
+                refreshProactively: true,
+                tokenRefresher: tokenRefresher
+            )
+            let credential = try CommunicationTokenCredential(withOptions: refreshOptions)
+            let options = AzureCommunicationChatClientOptions()
+            chatClient = try ChatClient(endpoint: appSettings.acsEndpoint, credential: credential, withOptions: options)
+        } catch {
+            //TODO: Add the code to do things when you failed to set up ChatClient
+        }
+    }
+
+
+    // MARK: UISceneSession Lifecycle
+    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
+        // Called when a new scene session is being created.
+        // Use this method to select a configuration to create the new scene with.
+        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
+    }
+
+    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
+        // Called when the user discards a scene session.
+        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
+        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
+    }
+
+    // MARK: Register for Push Notifications after the launch of App
+    func registerForPushNotifications() {
```

Comment:
Should this function be called in background task? We should refresh the registration in background tasks.